lib/Doctrine/DBAL/Portability/Statement.php

```diff
@@ -213,6 +213,55 @@ protected function fixRow($row, $iterateRow, $fixCase)
     }
 
     /**
+     * Fixes column values according to portability settings
+     *
+     * @param mixed[] $values Column values indexed by row number
+     * @param bool $emptyToNull Should empty strings be converted to NULL
+     * @param bool $rTrim Should strings be right-trimmed
+     * @return mixed[]
+     */
+    private function fixColumn(array $values, $emptyToNull, $rTrim)
+    {
+        if (!$emptyToNull && !$rTrim) {
```

Comment:
The most important thing is it saves a lot of complexity and function calls which only serve for 1 use-case. But as I said, not sure about that, it is just an idea that came to my mind.